src/sentry/tagstore/snuba/backend.py

```diff
@@ -677,11 +685,9 @@ def get_tag_value_paginator_for_projects(
             raise ValueError("Unsupported order_by: %s" % order_by)
 
         dataset = Dataset.Events
-        snuba_key = snuba.get_snuba_column_name(key)
-        if include_transactions and snuba_key.startswith("tags["):
-            snuba_key = snuba.get_snuba_column_name(key, dataset=Dataset.Discover)
-            if not snuba_key.startswith("tags["):
-                dataset = Dataset.Discover
+        if include_transactions:
+            dataset = Dataset.Discover
```

Comment:
We need to use the Discover dataset 100% of the time the include_transactions flag is passed. We cannot rely on transactions being present in Events, (they were only present by coincidence because they were temporarily being stored together with events) and they won't be in the future.